plugins/inputs/xtremio/xtremio.go

```diff
@@ -0,0 +1,324 @@
+	fields["ssds_read_iops"], _ = strconv.Atoi(gjson.Get(resp, "content.rd-iops").Str)
+	fields["ssds_write_bandwidth"], _ = strconv.Atoi(gjson.Get(resp, "content.wr-bw").Str)
+	fields["ssds_read_bandwidth"], _ = strconv.Atoi(gjson.Get(resp, "content.rd-bw").Str)
+	fields["ssds_num_bad_sectors"], _ = strconv.Atoi(gjson.Get(resp, "content.num-bad-sectors").Raw)
+
+	acc.AddFields("xio", fields, tags)
+}
+
+func (xio *XtremIO) GatherVolumes(wg *sync.WaitGroup, url string, acc telegraf.Accumulator) {
+	defer wg.Done()
+	resp, _ := xio.Call(url)
+	tags := map[string]string{
+		"guid":     gjson.Get(resp, "content.guid").Str,
+		"sys_name": gjson.Get(resp, "content.sys-name").Str,
+		"name":     gjson.Get(resp, "content.name").Str,
+	}
+	fields := make(map[string]interface{})
+	fields["volumes_read_iops"], _ = strconv.Atoi(gjson.Get(resp, "content.rd-iops").Str)
+	fields["volumes_write_iops"], _ = strconv.Atoi(gjson.Get(resp, "content.wr-iops").Str)
+	fields["volumes_read_latency"], _ = strconv.Atoi(gjson.Get(resp, "content.rd-latency").Str)
+	fields["volumes_write_latency"], _ = strconv.Atoi(gjson.Get(resp, "content.wr-latency").Str)
+	fields["volumes_data_reduction_ratio"], _ = strconv.ParseFloat(gjson.Get(resp, "content.data-reduction-ratio").Str, 64)
+	fields["volumes_provisioned_space"], _ = strconv.Atoi(gjson.Get(resp, "content.vol-size").Str)
+	fields["volumes_used_space"], _ = strconv.Atoi(gjson.Get(resp, "content.logical-space-in-use").Str)
+
+	acc.AddFields("xio", fields, tags)
+}
+
+func (xio *XtremIO) GatherXMS(wg *sync.WaitGroup, url string, acc telegraf.Accumulator) {
+	defer wg.Done()
+	resp, _ := xio.Call(url)
+	tags := map[string]string{
+		"guid":    gjson.Get(resp, "content.guid").Str,
+		"name":    gjson.Get(resp, "content.name").Str,
+		"version": gjson.Get(resp, "content.version").Str,
+		"xms_ip":  gjson.Get(resp, "content.xms-ip").Str,
+	}
+	fields := make(map[string]interface{})
+	fields["xms_write_iops"], _ = strconv.Atoi(gjson.Get(resp, "content.wr-iops").Str)
+	fields["xms_read_iops"], _ = strconv.Atoi(gjson.Get(resp, "content.rd-iops").Str)
+	fields["xms_overall_efficiency_ratio"], _ = strconv.ParseFloat(gjson.Get(resp, "content.overall-efficiency-ratio").Str, 64)
+	fields["xms_ssd_space_in_use"], _ = strconv.Atoi(gjson.Get(resp, "content.ssd-space-in-use").Str)
+	fields["xms_ram_in_use"], _ = strconv.Atoi(gjson.Get(resp, "content.ram-usage").Str)
+	fields["xms_ram_total"], _ = strconv.Atoi(gjson.Get(resp, "content.ram-total").Str)
+	fields["xms_cpu_usage_total"], _ = strconv.ParseFloat(gjson.Get(resp, "content.cpu").Raw, 64)
+	fields["xms_write_latency"], _ = strconv.Atoi(gjson.Get(resp, "content.wr-latency").Str)
+	fields["xms_read_latency"], _ = strconv.Atoi(gjson.Get(resp, "content.rd-latency").Str)
+	fields["xms_user_accounts_count"], _ = strconv.Atoi(gjson.Get(resp, "content.num-of-user-accounts").Raw)
+
+	acc.AddFields("xio", fields, tags)
+}
+
+func (xio *XtremIO) Call(endpoint string) (string, error) {
```

Comment:
Do not export this function
```suggestion
func (xio *XtremIO) call(endpoint string) (string, error) {
```